action/entrypoint.sh

```
@@ -136,29 +136,44 @@ else:
-echo "Findings: ${FINDINGS_COUNT}"
-
 # ─── Generate badge JSON (shields.io endpoint format) ────────────────────────
 
 BADGE_LABEL="${INPUT_BADGE_LABEL:-foxguard}"
 BADGE_FILE="${RUNNER_TEMP:-/tmp}/foxguard-badge.json"
 
-if [ "${FINDINGS_COUNT}" = "0" ]; then
+if [ -z "${FINDINGS_COUNT}" ] && [ "${EXIT_CODE}" -eq 2 ]; then
+    BADGE_MESSAGE="error"
+    BADGE_COLOR="red"
+elif [ "${FINDINGS_COUNT}" = "0" ] || ( [ -z "${FINDINGS_COUNT}" ] && [ "${EXIT_CODE}" -eq 0 ] ); then
```

Shellcheck (0.11.0)

[style] Use { ..; } instead of (..) to avoid subshell overhead.

(SC2235)